dockstore-common/src/main/java/io/dockstore/common/DescriptorLanguageSubclass.java

```diff
@@ -83,4 +84,8 @@ public static DescriptorLanguageSubclass convertShortNameStringToEnum(final Stri
             .filter(subclass -> subclass.getShortName().equalsIgnoreCase(descriptorSubclass)).findFirst();
         return first.orElseThrow(() -> new UnsupportedOperationException("language not supported yet"));
     }
+
+    public static Set<DescriptorLanguageSubclass> valuesForEntryType(EntryType type) {
```

Comment:
May be premature optimisation, but could probably just be stored in constructor(s) since it doesn't change.

Reply:
Probably don't need to optimize this, the amount of compute consumed is miniscule.  Also, if we compute this in the `EntryType` constructor (which is what you are proposing, I think) we'd have to resolve the problem of the circular references between the `EntryType` and `DescriptorLanguageSubclass` enums, which will be a bit icky.

Reply:
I meant the constructor for this class.